azurelinuxagent/common/osutil/default.py

```diff
@@ -267,11 +276,22 @@ def enable_firewall(self, dst_ip, uid):
 
         try:
             wait = self.get_firewall_will_wait()
-
             # If the DROP rule exists, make no changes
             try:
                 drop_rule = _get_firewall_drop_command(wait, "-C", dst_ip)
                 shellutil.run_command(drop_rule)
+                try:
```

Comment:
This should not be done here... this code is just checking whether the firewall rule is already setup (by checking for the drop rule). As the comment @ line 279 points out, if the firewall is already setup, this code should make no changes.

Reply:
Moved the logic to add the new rule under Update.py